background.js

```diff
@@ -83,17 +83,19 @@ function formatParams(params) {
 }
 
 function rightClickSearch(word){
-    console.log("Sending request..");
-    searchSpotify(word);
-    chrome.tabs.create({
-        url: chrome.extension.getURL('popup-signed-in.html'),
-        active: false
-    }, function(tab) {
-        // After the tab has been created, open a window to inject the tab
-        chrome.windows.create({
-            tabId: tab.id,
-            type: 'popup',
-            focused: true
+    //Store the word in storage, so it can be accessed by the frontend.
+    chrome.storage.sync.set({ "last_search": word }, function () {
+        console.log("Last Query saved.");
+        chrome.tabs.create({
+            url: chrome.extension.getURL('popup-signed-in.html'),
+            active: false
+        }, function(tab) {
+            // After the tab has been created, open a window to inject the tab
+            chrome.windows.create({
+                tabId: tab.id,
+                type: 'popup',
+                focused: true
+            });
         });
     });
 }
```

Comment:
This is a smart workaround! However, I have been working on this simultaneously on my local, instead of opening this pop-up, what I figured we could do was:
- Highlight text on webpage --> Right click --> Open context menu --> Select SPOT 
- We'll send just the data that was highlighted to the extension, probably trigger a notification on the bottom right corner (not sure if this is feasible - but looking into it)
- User will manually open the SPOT extension
- Search results of most recent highlighted text should be present

does this make sense?
I'll push these changes to a new branch, and  have to review them if you'd like. Post which we can figure what is the most optimum way of proceeding with this functionality!

Reply:
Absolutely, the notification idea sounds great.  The new window was making the css quite grotesque anyway.

Reply:
Truu haha, will push to the new branch in a bit and ping you for a review!!